File: tests/test_models.py
from ss.models import Index, Range

def test_parse_roundtrip():
    for row in range(50):
        for col in range(50):
            i = Index(row=row, col=col)
            s = str(i)
            assert Index.parse(s) == i

def test_range_denormalized():
    assert Range.parse('ZZ1:A10') == Range(
        Index.parse('A1'),
        Index.parse('ZZ10')
    )

File: javascript/test/test_models.js
let models = require("../sheet/models.js");
let assert = require("assert");

describe("index", () => {
  for (let [row, col, label] of [
       [0, 0, 'A1'],
       [0, 1, 'B1'],
       [1, 0, 'A2'],
       [123, 28, 'CC124']
  ]) {
    it('Knows its label', () => {
      index = new models.Index(row, col)
      assert.equal(index.label, label)
    });

    it('Parses correctly', () => {
      index = models.Index.parse(label)
      assert.equal(index.row, row)
      assert.equal(index.col, col)
    });
  }
});
